zcash_primitives/src/transaction/mod.rs

```diff
@@ -732,6 +760,9 @@ impl Transaction {
         let sapling_bundle = sapling_serialization::read_v5_bundle(&mut reader)?;
         let orchard_bundle = orchard_serialization::read_v5_bundle(&mut reader)?;
 
+        #[cfg(all(zcash_unstable = "nu7", feature = "zip-233"))]
+        let zip233_amount = Zatoshis::ZERO;
+
         #[cfg(zcash_unstable = "zfuture")]
```

Comment:
I suggest deciding whether ZFuture is based on v5 or v6 and only including it in that method, rather than including it in both (resulting in dead code).